### ISSUE TITLE
Crash when folding code

### ISSUE THREAD

**Issue by jemdev** [microsoft/vscode#214544]
Type: <b>Bug</b>

Since the last update, when I fold manually my code (any language, js, PHP, HTML, etc..), for example, I fold a part of my code, the UI become white and VS Code crash and disappear within 3 or 4 seconds. 

VS Code version: Code 1.90.0 (89de5a8d4d6205e5b11647eb6a74844ca23d2573, 2024-06-04T19:43:07.605Z)
OS version: Linux x64 5.4.0-182-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7300HQ CPU @ 2.50GHz (4 x 1680)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|6, 3, 3|
|Memory (System)|15.39GB (7.37GB free)|
|Process Argv|--no-sandbox --force-user-env --crash-reporter-id 5eba499c-ee80-49fd-b721-b550a96f0676|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (30)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-intelephense-client|bme|1.10.4
cookiejs|cpa|1.4.0
profiler-php-vscode|DEV|1.47.15512
githistory|don|0.6.20
jquerysnippets|don|0.0.1
apacheconf-snippets|eim|1.3.0
prettier-vscode|esb|10.4.0
codespaces|Git|1.16.22
remotehub|Git|0.62.0
vscode-pull-request-github|Git|0.88.1
live-sass|gle|6.1.2
todo-tree|Gru|0.0.226
google-translate|han|1.2.9
svn-scm|joh|2.17.0
twig-language-2|mbl|0.9.4
rainbow-csv|mec|3.12.0
easy-jquery|Moh|1.3.0
vscode-docker|ms-|1.29.1
vscode-language-pack-fr|MS-|1.89.2024050109
remote-containers|ms-|0.369.0
remote-repositories|ms-|0.40.0
format-html-in-php|rif|1.7.0
phpdoc-generator|ron|1.1.1
markdown-preview-enhanced|shd|0.8.13
html5-boilerplate|sid|1.1.1
pdf|tom|1.2.2
vscode-phpsab|val|0.0.18
php-debug|xde|1.34.0
php-pack|xde|1.0.3
php-intellisense|zob|1.3.3

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
dsvsc021:30996838
0ee40948:31013168
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupyter:31046869
26j00206:31048877

```

</details>

**Comment by jemdev** [comment]
No, it started happening after the last VS Code update and it only crashes when I try to fold or unfold code. Otherwise, VS Code works very correctly.

**Comment by gjsjohnmurray** [comment]
Out of interest, do any of the workarounds in that issue help you, apart from the revert one?

**Comment by gjsjohnmurray** [comment]
Also, please try the Start Extension Bisect command.

**Comment by jemdev** [comment]
I'll try that very soon, thanks for the idea

**Comment by Anuraj0124** [comment]
I have the same scenario. I updated the VS Code a few days back. I tried the Start Extension Bisect command. It gave this pop-up as I was able to reproduce the issue, even if all extensions were disabled.

The Application crashes at random, when I scroll over the collapse and expand code column, or click on the actions.
![image](https://github.com/microsoft/vscode/assets/56624675/91f4d33e-86f9-40e0-802e-4333cafe753b)

**Comment by iamshouvikmitra** [comment]
Here's my dump.zip
[dump.zip](https://github.com/user-attachments/files/15803834/dump.zip)

**Comment by Anuraj0124** [comment]
@jemdev @counselme @iamshouvikmitra 

You can do this if you have installed the snap version. This fixed my issue.

Solution
```
1. I uninstalled the snap version

2. Downloaded https://update.code.visualstudio.com/1.90.0/linux-deb-x64/stable

3. then run `sudo dpkg -i code_1.90.0-1717531825_amd64.deb`
```

**Comment by iamshouvikmitra** [comment]
Yeah, the issue is in snap versoin
